evoting/service/service.go

```diff
@@ -174,6 +179,47 @@ func (s *Service) Login(req *evoting.Login) (*evoting.LoginReply, error) {
 	return &evoting.LoginReply{Token: token, Admin: admin, Elections: elections}, nil
 }
 
+// LookupSciper calls https://people.epfl.ch/cgi-bin/people/vCard?id=sciper
+// to convert scipers to names
+func (s *Service) LookupSciper(req *evoting.LookupSciper) (*evoting.LookupSciperReply, error) {
+	url := fmt.Sprintf("https://people.epfl.ch/cgi-bin/people/vCard?id=%s", req.Sciper)
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	reply := &evoting.LookupSciperReply{}
+	search := regexp.MustCompile("[:;]")
+	for _, line := range strings.Split(string(body), "\n") {
+		fr := search.Split(line, 2)
+		if len(fr) != 2 {
+			continue
+		}
+		value := strings.Replace(fr[1], "CHARSET=UTF-8:", "", 1)
```

Comment:
If fr[1] does not start with CHARSET=UTF-8, return "error unknown character encoding" to help the next poor sucker debug what went wrong.

Reply:
Not all fields are of that charset type:
```
BEGIN:VCARD
VERSION:2.1
FN;CHARSET=UTF-8:Linus Gasser
N;CHARSET=UTF-8:Gasser;Linus;;;
ADR;TYPE=WORK;CHARSET=UTF-8:EPFL IC IINFCOM DEDIS ; BC 208 (Bâtiment BC) ; Station 14 ; CH-1015 Lausanne
EMAIL;TYPE=INTERNET:linus.gasser@epfl.ch
URL:https://people.epfl.ch/linus.gasser
TITLE;CHARSET=UTF-8:
TEL;TYPE=WORK:+41216936770 
ORG:EPFL
END:VCARD
```